pkg/network/common/egressip_test.go

```diff
@@ -362,38 +364,38 @@ func TestMultipleNamespaceEgressIPs(t *testing.T) {
 		t.Fatalf("%v", err)
 	}
 
-	// Now assigning that IP to a node should switch OVS to use that since it's first in the list
+	// Now assigning that IP to a node should cause ovs to add both of them
 	updateHostSubnetEgress(eit, &networkv1.HostSubnet{
 		HostIP:    "172.17.0.4",
 		EgressIPs: []networkv1.HostSubnetEgressIP{"172.17.0.101"},
 	})
 	err = w.assertChanges(
 		"claim 172.17.0.101 on 172.17.0.4 for namespace 42",
+		"namespace 42 via 172.17.0.100 on 172.17.0.3",
 		"namespace 42 via 172.17.0.101 on 172.17.0.4",
 	)
 	if err != nil {
 		t.Fatalf("%v", err)
 	}
 
-	// Swapping the order in the NetNamespace should swap back
+	// Swapping the order in the NetNamespace should do nothing since we are using both of them
 	updateNetNamespaceEgress(eit, &networkv1.NetNamespace{
 		NetID:     42,
 		EgressIPs: []networkv1.NetNamespaceEgressIP{"172.17.0.100", "172.17.0.101"},
 	})
-	err = w.assertChanges(
-		"namespace 42 via 172.17.0.100 on 172.17.0.3",
-	)
+	err = w.assertNoChanges()
 	if err != nil {
 		t.Fatalf("%v", err)
 	}
 
-	// Removing the inactive egress IP from its node should have no effect
+	// Removing the 1 egress IP from its node should cause it to be removed
```

Comment:
hm... the new rules are kind of ruining the logic of the unit test... It may be worth going through it again and thinking about if there are any cases we could be testing better now. And if not, then at least some parts of this are basically redundant now and could be dropped.